contracts/rewards/src/contract/execute.rs

```diff
@@ -0,0 +1,211 @@
+    #[test]
+    fn test_get_current_epoch_next_epoch() {
+        let (mut contract, current_epoch, stored_params) = setup();
+        let epoch_duration: u64 = stored_params.params.epoch_duration.into();
+
+        // epoch should increase by one
+        let block = current_epoch.block_height_started + epoch_duration;
+        let new_epoch = contract.get_current_epoch(block).unwrap();
+        assert_ne!(new_epoch, current_epoch);
+        assert_eq!(
+            Epoch {
+                epoch_num: 2,
+                block_height_started: block,
+                rewards: current_epoch.rewards.clone()
+            },
+            new_epoch
+        );
+
+        // epoch should increase by one, but start of epoch is in the past
+        let block = current_epoch.block_height_started + epoch_duration + epoch_duration / 2;
+        let new_epoch = contract.get_current_epoch(block).unwrap();
+        assert_ne!(new_epoch, current_epoch);
+        assert_eq!(
+            Epoch {
+                epoch_num: 2,
+                block_height_started: block - epoch_duration / 2,
+                rewards: current_epoch.rewards.clone()
+            },
+            new_epoch
+        );
+    }
+
+    /// Tests that the current epoch is computed correctly when a number of blocks greater than 2 times the epoch duration passes.
+    /// The epoch number should increase by more than one.
+    #[test]
+    fn test_get_current_epoch_skip_epochs() {
+        let (mut contract, current_epoch, stored_params) = setup();
+        let epoch_duration: u64 = stored_params.params.epoch_duration.into();
+
+        // epoch should increase by more than one
+        let block = current_epoch.block_height_started + epoch_duration * 4;
+        let new_epoch = contract.get_current_epoch(block).unwrap();
+        assert_ne!(new_epoch, current_epoch);
+        assert_eq!(
+            Epoch {
+                epoch_num: 5,
+                block_height_started: block,
+                rewards: current_epoch.rewards.clone()
+            },
+            new_epoch
+        );
+
+        // epoch should increase by more than one, but start of epoch is in the past
+        let block = current_epoch.block_height_started + epoch_duration * 4 + epoch_duration / 2;
+        let new_epoch = contract.get_current_epoch(block).unwrap();
+        assert_ne!(new_epoch, current_epoch);
+        assert_eq!(
+            Epoch {
+                epoch_num: 5,
+                block_height_started: block - epoch_duration / 2,
+                rewards: current_epoch.rewards.clone()
+            },
+            new_epoch
+        );
+    }
+
+    fn create_contract(stored_params: StoredParams) -> Contract<state::MockStore> {
+        let mut store = state::MockStore::new();
+        store
+            .expect_load_params()
+            .returning(move || stored_params.clone());
+        Contract { store }
+    }
+
+    fn setup() -> (Contract<state::MockStore>, Epoch, StoredParams) {
```

Comment:
I'd prefer to have parameters for duration, epoch_num and block_height_started, so they are clearly visible in the tests